### PR TITLE
Feature/26-builder service

### PR DESCRIPTION
**Pull request by pashamed**
https://trello.com/c/sl0VdNhm/26-builder-service

### CODE REVIEW

File: .github/workflows/сontinuous-integration.yml [hunk 1]
```diff
@@ -2,24 +2,26 @@ name: Continuous Integration
...
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: 6.0.x
+      - name: Restore .NET dependencies
+        run: dotnet restore backend/LeetWars.All.sln
+      - name: Build services
+        run: dotnet build backend/LeetWars.BuilderService/LeetWars.BuilderService.csproj
+      - name: Build .NET
+        run: dotnet build backend/LeetWars.All.sln --no-restore
```
Comment (tatianahutii) [inline .github/workflows/сontinuous-integration.yml:24]: you shouldn't change this

File: backend/LeetWars.BuilderService/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,2 @@
+FROM scratch
+WORKDIR /
```
Comment (tatianahutii) [inline backend/LeetWars.BuilderService/Dockerfile:2]: it should not be empty

File: docker-compose.apps.yml [hunk 3]
```diff
@@ -1,53 +1,61 @@
...
-    networks:
-        - back
-        - front
+    leetwars_builder:
+        build:
+            context: ./backend
+            dockerfile: ./LeetWars.BuilderService/Dockerfile
+        container_name: 'leetwars_builder'
+        environment:
+            ASPNETCORE_ENVIRONMENT: 'Development'
```
Comment (tatianahutii) [inline docker-compose.apps.yml:40]: do the same as in services above

File: backend/LeetWars.BuilderService/LeetWars.BuilderService.csproj [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net6.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+  </PropertyGroup>
+
+</Project>
```
Comment (tatianahutii) [inline backend/LeetWars.BuilderService/LeetWars.BuilderService.csproj:9]: your service name should be just Builder
and should contain more files, and implement one of the architectures (Minimal API, The Layered Architectural, CQRS)

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: backend/LeetWars.BuilderService/Dockerfile (hunk 2 above)
Comment (pashamed) [reply backend/LeetWars.BuilderService/Dockerfile:2]: it should build other backend images ?

Comment (tatianahutii) [reply backend/LeetWars.BuilderService/Dockerfile:2]: no, all Dockerfiles should have the same structure and build only inself

**Comment by tatianahutii** [comment]
Branch should contains cars number

File: .github/workflows/сontinuous-integration.yml (hunk 1 above)
Comment (pashamed) [reply .github/workflows/сontinuous-integration.yml]: removed

File: backend/.dockerignore [hunk 5]
```diff
@@ -0,0 +1,25 @@
+**/.classpath
```
Comment (tatianahutii) [inline backend/.dockerignore:1]: We already have this file

File: backend/Builder/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+FROM mcr.microsoft.com/dotnet/aspnet:6.0 AS base
+ARG PROJECT_PORT
+WORKDIR /app
+EXPOSE $PROJECT_PORT
+
+FROM mcr.microsoft.com/dotnet/sdk:6.0 AS build
+WORKDIR /src
+COPY ["Builder/Builder.csproj", "Builder/"]
+RUN dotnet restore "Builder/Builder.csproj"
```
Comment (tatianahutii) [inline backend/Builder/Dockerfile:13]: LeetWars.Builder =)

File: backend/LeetWars.Notifier/Dockerfile [hunk 7]
```diff
@@ -2,6 +2,7 @@ FROM mcr.microsoft.com/dotnet/aspnet:6.0-focal AS base
 ARG PROJECT_PORT
 WORKDIR /app
 EXPOSE $PROJECT_PORT
+EXPOSE 80
```
Comment (tatianahutii) [inline backend/LeetWars.Notifier/Dockerfile:5]: why did you add this here?

File: docker-compose.apps.yml [hunk 8]
```diff
@@ -1,53 +1,68 @@
...
-        - "4200:80"
-    networks:
-        - back
-        - front
+    leetwars_builder:
+        build:
+            context: ./backend
+            dockerfile: ./Builder/Dockerfile
+            args:
+                - PROJECT_PORT=5090
```
Comment (tatianahutii) [inline docker-compose.apps.yml:39]: add to the Readme file

File: docker-compose.apps.yml [hunk 9]
```diff
@@ -1,53 +1,68 @@
...
+        ports:
+            - "5090:5090"
+        networks:
+            - back
+        environment:
+            ASPNETCORE_ENVIRONMENT: 'Development'
+
+    leetwars_client:
+        depends_on:
+            - leetwars_core
```
Comment (tatianahutii) [inline docker-compose.apps.yml:51]: add builder to depends

**Review by tatianahutii — CHANGES_REQUESTED** [review]

**Review by tatianahutii — APPROVED** [review]

**Review by apofis96 — APPROVED** [review]
